test/integration/proxy_proto_integration_test.cc

```diff
@@ -314,30 +314,33 @@ ProxyProtoFilterChainMatchIntegrationTest::ProxyProtoFilterChainMatchIntegration
   });
 }
 
+void ProxyProtoFilterChainMatchIntegrationTest::send(const std::string& data) {
+  initialize();
+
+  // Set verify to false because it is expected that Envoy will immediately disconnect after
+  // receiving the PROXY header, and it is a race whether the `write()` will fail due to
+  // disconnect, or finish the write before receiving the disconnect.
+  constexpr bool verify = false;
```

Comment:
👍  Looks good and a question from me is that:
https://github.com/envoyproxy/envoy/blob/main/test/integration/integration_tcp_client.cc#L142
It seems setting verify to false will skip the writting status and unexpected disconnection check and directly skip the following part?https://github.com/envoyproxy/envoy/blob/main/test/integration/integration_tcp_client.cc#L123-L128